pyomo/solvers/plugins/solvers/SCIPAMPL.py

```diff
@@ -288,115 +288,118 @@ def _postsolve(self):
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.stoppedByLimit
 
         # OK # other='other' # Other, uncategorized normal termination
 
-        elif results.solver.message == "memory limit reached":
+        elif "memory limit reached" in results.solver.message:
             results.solver.status = SolverStatus.ok
             results.solver.termination_condition = TerminationCondition.other
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.stoppedByLimit
 
         # OK # other='other' # Other, uncategorized normal termination
 
-        elif results.solver.message == "gap limit reached":
+        elif "gap limit reached" in results.solver.message:
             results.solver.status = SolverStatus.ok
             results.solver.termination_condition = TerminationCondition.other
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.stoppedByLimit
 
         # OK # other='other' # Other, uncategorized normal termination
 
-        elif results.solver.message == "solution limit reached":
+        elif "solution limit reached" in results.solver.message:
             results.solver.status = SolverStatus.ok
             results.solver.termination_condition = TerminationCondition.other
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.stoppedByLimit
 
         # OK # other='other' # Other, uncategorized normal termination
 
-        elif results.solver.message == "solution improvement limit reached":
+        elif "solution improvement limit reached" in results.solver.message:
             results.solver.status = SolverStatus.ok
             results.solver.termination_condition = TerminationCondition.other
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.stoppedByLimit
 
         # OK # optimal='optimal' # Found an optimal solution
 
-        elif results.solver.message == "optimal solution found":
+        elif "optimal solution" in results.solver.message:
             results.solver.status = SolverStatus.ok
             results.solver.termination_condition = TerminationCondition.optimal
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.optimal
-            if results.problem.sense == ProblemSense.minimize:
-                results.problem.lower_bound = results.solver.primal_bound
-            else:
-                results.problem.upper_bound = results.solver.primal_bound
+            try:
+                if results.problem.sense == ProblemSense.minimize:
+                    results.problem.lower_bound = results.solver.primal_bound
+                else:
+                    results.problem.upper_bound = results.solver.primal_bound
+            except AttributeError:
```

Comment:
@michaelbynum would you mind adding a short comment here explaining why this is needed or when you would end up here? Is this from a difference in solver versions?

Reply:
Done. Sorry - I was in a hurry yesterday.

Reply:
Sorry, I only added comments to the PR. Now, I've added comments to the code.